### PR TITLE
Expand CreateSpan in T0

### PR DESCRIPTION
**Pull request by MichalPetryka**
I've noticed my app allocate 100MB of `RuntimeFieldInfoStub` instances in the VM implementation while running the T0 code in my app, expanding in T0 should avoid such allocations.

### CODE REVIEW

**Comment by MichalPetryka** [comment]
@jkotas Could this get a backport to 8.0?

**Comment by jkotas** [comment]
It is up to the JIT team to make the call on a backport and bring it through the servicing process.

I do not think that this meets the bar for backport to .NET 8. It is by design that T0 compilation consumes extra memory, including extra GC allocations that would be eliminated by full JIT optimizations.

> I've noticed my app allocate 100GB of RuntimeFieldInfoStub instances in the VM implementation

Why is not T1 compilation kicking fast enough for your app? It takes a while to allocate 100GB of RuntimeFieldInfoStub, I would expect T1 compilation to kick in a long before this much of RuntimeFieldInfoStub instances get allocated.

**Comment by MichalPetryka** [comment]
> It is by design that T0 compilation consumes extra memory, including extra GC allocations that would be eliminated by full JIT optimizations.

I understand that but with this optimalization pessimizing T0 code so much using it without AggressiveOptimization turns out to be a bad idea, making it effectively useless and a serious performance issue.

> It takes a while to allocate 100GB of RuntimeFieldInfoStub

I've actually misread the data, it's able to allocate 100MB, not 100GB. The code in question consists of many hot loops which probably take a lot of time to OSR.

**Comment by EgorBo** [comment]
I agree with Jan that we already have tons of similar cases in tier0, e.g. `Enum.HasFlag` (already fixed), also everything related to devirtualizations of interfaces calls on structs when jit can see the exact type (see https://github.com/dotnet/runtime/issues/90965) and other patterns where only optimized code can get rid of allocations.

We also know that sometimes a method may stay in Tier0 for too long because of the "promotion queue" problem (We had an idea to make it priority-based and e.g. all methods with allocators inside could get a higher priority).

File: src/coreclr/jit/importercalls.cpp [hunk 1]
```diff
@@ -2551,6 +2551,9 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             // This one is just `return true/false`
             case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsKnownConstant:
 
+            // Not expanding this leads to huge allocation in T0
```
Comment (EgorBo) [inline src/coreclr/jit/importercalls.cpp:2554]: Can you make the comment a bit less drammatic? 🙂 I'm referring to 'huge" word

**Review by EgorBo — APPROVED** [review]

**Comment by JulieLeeMSFT** [comment]
@MichalPetryka, there are build errors. Please resolve it before merging.

**Comment by MichalPetryka** [comment]
@MihuBot -tier0

**Comment by EgorBo** [comment]
@MichalPetryka @MihaZupan is MihuBot failing because of this PR or it's a general issue? Does it fail on other changes?

**Comment by MihaZupan** [comment]
It's failing somewhere in `jit-diff`. I'll try it on different PRs now.

**Comment by MihaZupan** [comment]
Seems like the issue is specifically with this PR and tier 0.
It's working fine on other PRs, or on this PR without the `--tier0` flag.

**Comment by EgorBo** [comment]
Let's run some pipelines which don't disable TieredCompilation because green inner-loop CI likely means it was never executed in Tier0.

**Comment by EgorBo** [comment]
@MihuBot -tier0

**Comment by EgorBo** [comment]
I've ran jitdiff locally with --tier0 and this PR didn't seem to cause any issues for me so I am going to merge it.

**Pull request merged**